# gazebo_ros_control:
#   pid_gains:
#     front_left_wheel: {p: 100.0, i: 0.01, d: 10.0}
#     front_right_wheel: {p: 100.0, i: 0.01, d: 10.0}
gazebo_ros_control:
  pid_gains:
    front_left_wheel:
      p: 200
      i: 200
      d: 0
    front_right_wheel:
      p: 200
      i: 200
      d: 0

nadie:
  motor_controller:
    control_loop_hz: 50.0
    max_command_retries: 4
    max_seconds_uncommanded_travel: 0.5
    port_address: 128
    quad_pulses_per_meter: 5965.29
    quad_pulses_per_revolution: 2248.86
    usb_device_name: /dev/roboclaw
    wheel_radius: 0.06

  joint_state_controller:
    type: joint_state_controller/JointStateController
    publish_rate: 50

  # base_to_lwheel_velocity_controller:
  #   type: effort_controllers/JointVelocityController  # From URDF <hardwareInterface>hardware_interface/VelocityJointInterface</hardwareInterface>
  #   joint: base_to_lwheel_joint                       # From URDF <joint name="base_to_lwheel_joint" type="continuous">
  #   pid: {p: 8762.98571, i: 1535.49646, d: 0.0}       # From IonMotion setup program.

  # base_to_rwheel_velocity_controller:
  #   type: effort_controllers/JointVelocityController  # From URDF <hardwareInterface>hardware_interface/VelocityJointInterface</hardwareInterface>
  #   joint: base_to_rwheel_joint                       # From URDF <joint name="base_to_rwheel_joint" type="continuous">
  #   pid: {p: 9542.41265, i: 1773.65086, d: 0.0}       # From IonMotion setup program.

  # joint_names: [base_to_lwheel_joint, base_to_rwheel_joint]

  diff_drive_controller:
    type        : "diff_drive_controller/DiffDriveController"
    left_wheel  : 'front_left_wheel'
    right_wheel : 'front_right_wheel'
    publish_rate: 50.0
    pose_covariance_diagonal : [0.001, 0.001, 1000000.0, 1000000.0, 1000000.0, 1000.0]
    twist_covariance_diagonal: [0.001, 0.001, 1000000.0, 1000000.0, 1000000.0, 1000.0]

    # Wheel separation and diameter. These are both optional.
    # diff_drive_controller will attempt to read either one or both from the
    # URDF if not specified as a parameter
    wheel_separation : 0.305  Check URDF #####
    #wheel_radius : 0.0762

    # Wheel separation and radius multipliers
    wheel_separation_multiplier: 1.0 # default: 1.0
    wheel_radius_multiplier    : 1.0 # default: 1.0

    # Velocity commands timeout [s], default 0.5
    cmd_vel_timeout: 0.25

    # Base frame_id
    base_frame_id: base_link
    odom_frame_id: odom
    enable_odom_tf: true

    # Velocity and acceleration limits
    # Whenever a min_* is unspecified, default to -max_*
    linear:
      x:
        has_velocity_limits    : true
        max_velocity           : 0.7
        min_velocity           : -0.7
        has_acceleration_limits: true
        max_acceleration       : 3.0
        min_acceleration       : -3.0
    angular:
      z:
        has_velocity_limits    : true
        max_velocity           : 3.0
        has_acceleration_limits: true
        max_acceleration       : 3.0
